src/core/config/configErrors.ts

```diff
@@ -0,0 +1,52 @@
+import { DSNPError } from "../errors";
+
+/**
+ * ConfigError indicates that some piece of configuration is not set and that
+ * the current action cannot proceed without it.
+ */
+export class ConfigError extends DSNPError {
+  constructor(message: string) {
+    super(message);
+    this.name = "ConfigError";
+  }
+}
+
+/**
+ * MissingSignerConfigError indicates that the signer config is not set.
+ */
+export class MissingSignerConfigError extends ConfigError {
+  constructor() {
+    super("Signer is not set.");
+    this.name = "MissingSignerConfigError";
+  }
+}
+
+/**
+ * MissingProviderConfigError indicates that the provider config is not set.
+ */
+export class MissingProviderConfigError extends ConfigError {
+  constructor() {
+    super("Blockchain provider is not set.");
+    this.name = "MissingProviderConfigError";
+  }
+}
+
+/**
+ * MissingStoreConfigError indicates that the store config is not set.
+ */
+export class MissingStoreConfigError extends ConfigError {
+  constructor() {
+    super("Store adapter was not found");
+    this.name = "MissingStoreConfigError";
+  }
+}
+
+/**
+ * MissingFromIdConfigError indicates that the current from id is missing.
+ */
+export class MissingFromIdConfigError extends ConfigError {
+  constructor() {
+    super("No from id found. Please authenticate a handle.");
+    this.name = "MissingUFromIdConfigError";
+  }
+}
```

Comment:
We can discuss it more in the next SDK meeting, but I was thinking, since `setConfig` is the only config module method we actually use at the porcelain level, we might want to move the rest of the config back into a plumbing module here with these errors, and expose a separate config porcelain with the one method we actually expect the users to call.

Reply:
I'm not sure about where inside, but I like the idea of only externally exposing `getConfig` and `setConfig` (aka never exposing the other *Config methods outside of the sdk)